the-collab-lab/tcl-45-smart-shopping-list
# Conditionally render About button

## Description



This PR adds conditional rendering of buttons navigating to About page and back from it.

1. When there is no token in the Home screen the user can click the About button and navigate to the About page to learn more about the app. To navigate back to the Home screen the user can use the Back button near the page title.

2. When there is a token the user can click the About button in the top bar to navigate to the About page, to return back there is a button called List in the same place where the About button was. I intended to reduce the number of buttons in the top bar.

I also restyled the about page a little bit to make it easier to read by increasing the font size, adding margins, removing a piece of repeated text from the first section, as well as added links to mentors' names.

I refactored the Home screen slightly by replacing the red ghost image with another one of a better quality, removed the width property from all characters and the question mark icon so that their images are not squeezed on a narrower screen.

## Related Issue



Closes #71  Closes #70 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

No token:

<img width="680" alt="Screen Shot 2022-08-31 at 3 15 57 PM" src="https://user-images.githubusercontent.com/63440229/187785932-d421a7d2-0f61-4173-ae5b-1d845861cb8f.png">

With a token: 

<img width="918" alt="Screen Shot 2022-08-31 at 3 16 16 PM" src="https://user-images.githubusercontent.com/63440229/187785940-fb3d23ca-69be-411c-911c-65a094cb57b9.png">


### After
With a token:


<img width="1139" alt="Screen Shot 2022-08-30 at 10 40 41 PM" src="https://user-images.githubusercontent.com/63440229/187595255-3f8af53b-e5a0-4e68-a4c5-606fa102c846.png">

<img width="1106" alt="Screen Shot 2022-08-30 at 10 40 32 PM" src="https://user-images.githubusercontent.com/63440229/187595260-feb19f5c-b688-4e49-9f53-f27387f3be3a.png">

No token:

<img width="653" alt="Screen Shot 2022-08-31 at 3 10 07 PM" src="https://user-images.githubusercontent.com/63440229/187784873-de950ff1-70d8-44c9-9bb7-90b3c62e251d.png">

<img width="900" alt="Screen Shot 2022-08-31 at 3 10 13 PM" src="https://user-images.githubusercontent.com/63440229/187784890-ebfe2191-e45c-47b7-9723-f6e6cd5fa371.png">

## QA

1. Check out `NZ-conditionally-tender-list-button` (I mistyped the word "render" and yes, the branch name does say "tender" 😂) and run `npm start`
2. Test how the About button works, navigates, disappears with/without a token.
